### ISSUE TITLE
[Graph Docs]  Trigger API Doctor validation pipeline even when no changes have been made on the `api-reference` folder

### ISSUE THREAD

**Issue by millicentachieng** [microsoftgraph/microsoft-graph-devx-api#2135]
We’ve encountered cases where only the conceptual topics were updated in a PR, causing the API Doctor validation pipeline not to run. As a result, the SDLSources stage wasn’t triggered, leading to potential security vulnerabilities being detected only after the changes were merged into the main branch.

By removing this check, we can ensure that the pipeline runs regardless of whether the api-reference folder has been modified.
